### PR TITLE
[breaking change] fix 265, log should be log10, and add ln

### LINKED ISSUE #265: Discrepancy in the meaning of log function
**Describe the bug**

In [Postgres](https://www.postgresql.org/docs/current/functions-math.html) `log` is equivalent to `log10` while `ln` exists to represent natural logarithm. Currently the datafusion function isn't consistent with this.

**To Reproduce**

run `select log(10)`

**Expected behavior**

it should be =1 rather than >1; also there should be a dedicated `ln` function

**Additional context**

### PR DESCRIPTION
**Pull request by Jimexist**
# Which issue does this PR close?

We generally require a GitHub issue to be filed for all bug fixes and enhancements and this helps us generate change logs for our releases. You can link an issue to this PR using the GitHub syntax. For example `Closes #123` indicates that this PR will close issue #123.

Closes #265.

 # Rationale for this change
 Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.  

# What changes are included in this PR?

There is no need to duplicate the description in the issue here but it is sometimes worth providing a summary of the individual changes in this PR.

# Are there any user-facing changes?

If there are user-facing changes then we may require documentation to be updated before approving the PR.

If there are any breaking changes to public APIs, please add the `breaking change` label.

### CODE REVIEW

File: datafusion/src/logical_plan/expr.rs [hunk 1]
```diff
@@ -1086,9 +1086,9 @@ unary_scalar_expr!(Trunc, trunc);
 unary_scalar_expr!(Abs, abs);
 unary_scalar_expr!(Signum, signum);
 unary_scalar_expr!(Exp, exp);
-unary_scalar_expr!(Log, ln);
```
Comment (alamb) [inline datafusion/src/logical_plan/expr.rs:1089]: perhaps we should keep a `log` entry / alias to `log10` for postgres compatibility?

**Review by alamb — COMMENTED** [review]
I think this is a good fix personally. 

I recommend we add some sort of end to end sql test that runs these functions and compares the results, but otherwise 👍  from me. Any thoughts @jorgecarleitao or @seddonm1 ?

**Review by jorgecarleitao — APPROVED** [review]
I agree, @alamb . Postgres spec ftw 👍 

Thanks @Jimexist for this!

File: datafusion/src/logical_plan/expr.rs (hunk 1 above)
Comment (Jimexist) [reply datafusion/src/logical_plan/expr.rs:1089]: i guess it's kept at the https://github.com/apache/arrow-datafusion/pull/271/files#diff-97178da6c90bfcb5e3dc80f06248ac1aebdec5a0e0e1d8ea235e2f571b5badf8R227 level, not here.

Comment (Jimexist) [reply datafusion/src/logical_plan/expr.rs:1089]: here the macro defs shall not allow dups

Comment (seddonm1) [reply datafusion/src/logical_plan/expr.rs:1089]: Definitely in favour of Postgres behaviour. 👍

Comment (Jimexist) [reply datafusion/src/logical_plan/expr.rs:1089]: see https://github.com/apache/arrow-datafusion/pull/275

**Comment by Jimexist** [comment]
I've added a test suite to make it future proof: https://github.com/apache/arrow-datafusion/pull/275

**Review by alamb — APPROVED** [review]
I gave it a little spin and it looks good. Thank you @Jimexist !

```
> CREATE EXTERNAL TABLE t(a double) STORED AS CSV LOCATION '/tmp/foo.csv';
0 rows in set. Query took 0 seconds.
> select * from t;
+---------+
| a       |
+---------+
| 1       |
| 2       |
| 3       |
| 3.14159 |
+---------+
4 row in set. Query took 0 seconds.
> select log(a), log10(a) from t;
+---------------------+---------------------+
| log(a)              | log10(a)            |
+---------------------+---------------------+
| 0                   | 0                   |
| 0.3010299956639812  | 0.3010299956639812  |
| 0.47712125471966244 | 0.47712125471966244 |
| 0.4971495058611233  | 0.4971495058611233  |
+---------------------+---------------------+
```

**Pull request merged**